pkgs/tools/nix/nixos-render-docs/default.nix

```diff
@@ -0,0 +1,51 @@
+{ lib
+, stdenv
+, python3
+, python3Minimal
+}:
+
+let
+  # python3Minimal can't be overridden with packages on Darwin, due to a missing framework.
+  # Instead of modifying stdenv, we take the easy way out, since most people on Darwin will
+  # just be hacking on the Nixpkgs manual (which also uses make-options-doc).
+  python = ((if stdenv.isDarwin then python3 else python3Minimal).override {
+    self = python;
+    includeSiteCustomize = true;
+  });
+
+  # TODO add our own small test suite, maybe add tests for these deps to channels?
+  markdown-it-py-no-tests = python.pkgs.markdown-it-py.override {
+    disableTests = true;
+  };
+  mdit-py-plugins-no-tests = python.pkgs.mdit-py-plugins.override {
+    markdown-it-py = markdown-it-py-no-tests;
+    disableTests = true;
+  };
+in
+
+python.pkgs.buildPythonApplication {
+  pname = "nixos-render-docs";
+  version = "0.0";
+  format = "pyproject";
+
+  src = ./src;
```

Comment:
This being more of a project directory now, does it generate files that should be filtered and/or added to a local `.gitignore`?